candle-wasm-examples/whisper/src/bin/m.rs

```diff
@@ -0,0 +1,41 @@
+use candle_wasm_example_whisper::worker::{Decoder as D, ModelData};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct Decoder {
+    decoder: D,
+}
+
+#[wasm_bindgen]
+impl Decoder {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        weights: Vec<u8>,
+        tokenizer: Vec<u8>,
+        mel_filters: Vec<u8>,
+    ) -> Result<Decoder, JsError> {
+        let decoder = D::load(ModelData {
+            tokenizer,
+            mel_filters,
+            weights,
```

Comment:
potentially we could have `config: JsValue` here? and fetch `config.json` from the model file, but I need to figure out how to convert it to a `struct Config `

Reply:
Sounds like a great idea if we want to support multiple versions (the command line version of the whisper example does). You're probably looking for something like:
```rust
let config: Config = serde_json::from_str(&json_string)?)?
```
You can grep in the repo to find some examples of this being used, it mostly requires the `Config` struct to have the proper `#[derive(Serialize, Deserialize)]` annotation so that the serialization/deseralization codes are generated.

Reply:
Certainly happy to defer this to a future PR if you want. Also note that this is still using the very old version of the whisper model so I plan on updating it to the latest version used by the command line tool, hopefully this should improve performance significantly.